### DISCUSSION TITLE
What if nerve and cuff circumference are different?

### DISCUSSION THREAD

**Discussion by PALMS-gui** [wmglab-duke/ascent#19] [Q&A] [answered]
Hi, could you please explain the following...

A cuff by design is a rigid structure of predefined size\circumference. Then it is assumed to be applied to various nerves.
Imagine, that the nerve of (significantly) smaller circumference.

1. What will happen in real experiment? 
I would expect the cuff gets wrapped around the nerve while possible, and the rest of the cuff hangs "out there" or it gets wrapped for the second "lap" (on top of already existing cuff). Both ways I assume would not affect the stimulation or the processes inside the nerve.

2. What will happen in ASCENT? 
Will the cuff shrink or cut itself off? Can this be controlled or what do you suggest doing in such case?
Does it matter if the difference is small or large? 

Thank you

**Comment by thedannymarsh** [comment, marked as answer]
Hi PALMS-gui,

1. The geometry of the cuff interface on the nerve will depend on the style of the cuff. What you are referring to sounds like a “spiral” cuff design. We agree that the cuff components that are “shielded” by internal layers of insulation are likely to have less impact on fiber responses to insulation than those that are not. To convince yourself of this, as a first pass you could model the same cuff design with different insulation thicknesses. As a second pass, you could create your own cuff design with a layer of insulation between active contacts and the nerve. Other cuff designs are more like a tube with a longitudinal cut (i.e., like the [tube cuff primitive](https://wmglab-duke-ascent.readthedocs.io/en/latest/S16-Library-of-part-primitives-for-electrode-contacts-and-cuffs.html) in ASCENT). The parts for creating cuffs and electrodes in ASCENT will continue to grow as we (and potentially other users) implement more sophisticated cuff designs for projects.
2. Cuffs in ASCENT have a resting inner diameter. If the nerve sample is smaller than this diameter, the cuff will not shrink beyond this resting inner diameter, and will position according to your cuff shift mode (https://wmglab-duke-ascent.readthedocs.io/en/latest/S19-Cuff-placement-on-nerve.html). Most cuffs are parameterized such that they will expand to accommodate a nerve larger than their resting inner diameter. For nerve samples smaller than the cuff, you could either allow cuff shifting to place the contact close to the nerve, or modify the cuff preset json file to shrink the cuff resting inner diameter. 

Hope this helps! Have a great weekend!

**Reply by PALMS-gui** [reply]
thank you for the answer, I will need some time to digest and understand it better. 
What I got in one of the simulations when the nerve is smaller than the cuff: 



that is the space between square electrodes shrinked and they even started overlapping, whereas I expected something like in one of your pictures:
![afbeelding](https://user-images.githubusercontent.com/72300527/159878556-8c8e56ca-edb2-4605-8433-7763d2d61750.png)
Though, I must admit, in my model config, I didn't use any Cuff Shift options (it was set to None)

**Reply by thedannymarsh** [reply]
In the configuration file which defines your custom cuff, I would make sure you have defined the inner diameter of the cuff such that it has a minimum value. I.e., the cuff's inner radius value will be the minimum diameter, unless the nerve's size requires the cuff to expand, then it will be the new expanded diameter. I would look to Purdue.json as a simple cuff which has these properties. Also note that you can manually change the parameters in COMSOL to test your cuff without re-running the pipeline (you can change r_nerve in your parameters and rebuild your geometry, seeing how your cuff reacts.)
